pkg/serverless/logs/logs_collector.go

```diff
@@ -196,7 +196,10 @@ func (lc *LambdaLogsCollector) processMessage(
 	if lc.enhancedMetricsEnabled {
 		tags := tags.AddColdStartTag(lc.extraTags.Tags, lc.lastRequestID == lc.coldstartRequestID)
 		if message.logType == logTypeFunction && !lc.outOfMemory {
-			lc.outOfMemory = serverlessMetrics.GenerateEnhancedMetricsFromFunctionLog(message.stringRecord, message.time, tags, lc.demux)
+			lc.outOfMemory = serverlessMetrics.ContainsOutOfMemoryLog(message.stringRecord)
+			if lc.outOfMemory {
+				serverlessMetrics.GenerateEnhancedMetricsFromFunctionLog(message.time, tags, lc.demux)
+			}
```

Comment:
nit

I _think_ you can alternatively do 

```go
			if lc.outOfMemory = serverlessMetrics.ContainsOutOfMemoryLog(message.stringRecord) ; lc.outOfMemory {
				serverlessMetrics.GenerateEnhancedMetricsFromFunctionLog(message.time, tags, lc.demux)
			}
```

Reply:
🧹💨